inc/math/linalg.hpp

```diff
@@ -363,7 +363,9 @@ namespace exo
 			}
 
 			template <ssize_t MC>
-			Mat<S, R, MC> operator* (Mat<S, C, MC>& m)
+			Mat<S, R, MC> operator* (Mat<S, C, MC>& m) { return *this * std::move(m); }
```

Comment:
Are you sure you want to call std::move in these operators? If I called operator* and expected to be passing by reference, I wouldn't want my matrix to be invalidated. I actually don't see the point of using std::move in any of these operator functions since it doesn't seem like the intent is ever to take the values of the matrices so much as it is to use them.

Reply:
The intention is to allow you to pass either a reference, or a inline literal. Do you have a better suggestion?

Reply:
I'm not sure I understand what you mean inline literal, but wouldn't `(*this) * m` work? If you call move, I think an r value reference is returned. After the function returns, I don't think the caller can trust that the `m` it passed in. Essentially, I think m will be valid, but in an undefined state state afterwards. Could be wrong though. For example:
```
std::vector<int> v{1,2,3,4,5};
std::vector<int> v2(std::move(v)); // binds an rvalue reference to v
assert(v.empty());
```

Reply:
Ahh, I understand what you are saying. I think the reason this stuff works is because of how the Mat and Vec types are implemented. Each simply has a statically allocated array of type S who's type and size are defined by template parameters. You literally can't move these. So normal move semantics don't break it. When I talk about literals, I mean having the ability to do something like this.
```
Vec<float, 3> x = M * { 1, 2, 3 };
```
This frequently makes math more semantic and clear. The other option here would be to have an overload like this.
```
Mat<S, R, MC> operator* (Mat<S, C, MC> m)
```
But I don't like that much, because then a full copy of m is made when the function is invoked. What do you think?

Reply:
In this light, I agree that the moves are useful. Might be better to instantiate them as std::arrays. Then the data can be swapped and the && functions work in the normal way. I still don't think we want to call move in in the & function though.